chunky/src/java/se/llbit/chunky/renderer/scene/AsynchronousSceneManager.java

```diff
@@ -74,15 +76,13 @@ public void setOnChunksLoaded(Runnable onChunksLoaded) {
   @Override public void run() {
     try {
       while (!isInterrupted()) {
-        synchronized (this) {
-          while (currentTask == null) {
-            wait();
-          }
-        }
-        currentTask.run();
-        synchronized (this) {
-          currentTask = null;
+        synchronized (taskQueue) {
+          while (taskQueue.isEmpty())
+            taskQueue.wait();
         }
+        Runnable task = taskQueue.poll();
```

Comment:
Could the explicit wait and locking logic be eliminated by using a `LinkedBlockingQueue` and `Runnable task = taskQueue.take()`?